internal/resourcemapping/resourcemapping.go

```diff
@@ -208,6 +208,17 @@ func ResourceAttributesToMonitoredResource(attrs ReadOnlyAttributes) *GceResourc
 	return mr
 }
 
+// getK8sClusterOrCloudPlatform returns "k8s_cluster" if K8SClusterNameKey is set
+// (which identifies cloud provider clusters like GKE, EKS, AKS, or non-cloud clusters like minikube)
+// Otherwise, it returns the cloud.platform value.
+func getK8sClusterOrCloudPlatform(attrs ReadOnlyAttributes) string {
+	if _, cluster := attrs.GetString(string(semconv.K8SClusterNameKey)); cluster {
+		return k8sCluster
```

Comment:
nit: kind of strange to mix platform semantic conventions and other strings.  Maybe return semconv.CloudPlatformGCPKubernetesEngine.Value.AsString() here instead?

Reply:
Yeah, I thought about that too. I don't think it's perfect either way (felt weird to return the platform as GCP when it could be any k8s).

It's odd to me that there isn't an attribute to indicate generic K8S platform. The best you can do, as far as I can tell, is just infer if `k8s.cluster.name` is set